frontend/app/js/custom.js

```diff
@@ -0,0 +1,18 @@
+$('#git_tree_content').on("select_node.jstree", function(e, data){
+  var git_node = data.node.data;
+  if(git_node.type === "jstree-file"){
+  $.ajax('blob/content', {
+    type: 'get',
+    data: { sha: git_node.sha , key: key},
+    success : function(response) {
+      $("#git_code").html(response.data);
+    },
+    error: function() {
+      alert("Error");
```

Comment:
put more descriptive message if possible, send the error response from error, when there is exception in server side like git timeout or invalid sha something..